### DISCUSSION TITLE
Best way to handle adding zlib support to openssl@1.1?

### DISCUSSION THREAD

**Discussion by dsXLII** [Homebrew/discussions#1353] [Everyday usage] [answered]
I regularly use the 'sslscan' application, which can be compiled to link to zlib. (zlib isn't required but adds functionality.) It's easy enough to modify the openssl@1.1 formula to enable zlib support - just replace "no-zlib" with "zlib-dynamic" and update a comment - and while this works for my one specific use case, I have no way of knowing whether adding zlib support would impact any of the (dozens? hundreds?) of other applications that link to openssl.

Is there a reason that Homebrew's openssl formula explicitly disables zlib support? If not, or if that reason is historic, do we want to change the openssl formula to enable zlib?

For my specific use case, there's a possible workaround, though one that's a bit more complicated. sslscan's built-in Makefile can download and build OpenSSL itself, and statically link to that version. (That option is available in case the system-wide openssl disables SSLv3/TLSv1.0/etc.) It's probably possible to update this 'internal' openssl configuration to link to zlib. That would be harder :) and would only work for that specific application.

**Comment by carlocab** [comment, marked as answer]
I think adding `zlib` support would be fine. Please open a PR for this so the change can be reviewed and tested properly, and include the following changes:
```diff
diff --git a/Formula/openssl@1.1.rb b/Formula/openssl@1.1.rb
index 3fa43ec1db0..5808f4c4b56 100644
--- a/Formula/openssl@1.1.rb
+++ b/Formula/openssl@1.1.rb
@@ -5,6 +5,7 @@ class OpensslAT11 < Formula
   mirror "https://www.mirrorservice.org/sites/ftp.openssl.org/source/openssl-1.1.1k.tar.gz"
   sha256 "892a0875b9872acd04a9fde79b1f943075d5ea162415de3047c327df33fbaee5"
   license "OpenSSL"
+  revision 1
   version_scheme 1
 
   livecheck do
@@ -21,6 +22,8 @@ class OpensslAT11 < Formula
 
   keg_only :shadowed_by_macos, "macOS provides LibreSSL"
 
+  uses_from_macos "zlib"
+
   on_linux do
     resource "cacert" do
       # homepage "http://curl.haxx.se/docs/caextract.html"
```
Please also make sure to use homebrew/core style for your commit message. Thanks!
